subsys/logging/log_mgmt.c

```diff
@@ -358,9 +358,12 @@ void z_log_runtime_filters_init(void)
 int log_source_id_get(const char *name)
 {
 	for (int i = 0; i < log_src_cnt_get(Z_LOG_LOCAL_DOMAIN_ID); i++) {
-		if (strcmp(log_source_name_get(Z_LOG_LOCAL_DOMAIN_ID, i),
-			   name) == 0) {
-			return i;
+		const char *sname = log_source_name_get(Z_LOG_LOCAL_DOMAIN_ID, i);
+
+		if (sname) {
+			if (strcmp(sname, name) == 0) {
+				return i;
+			}
```

Comment:
Nit: single `if` statement?
```suggestion
		if (sname && strcmp(sname, name) == 0) {
			return i;
```